DetourCrowd/Source/DetourObstacleAvoidance.cpp

```diff
@@ -331,7 +331,7 @@ float dtObstacleAvoidanceQuery::processSample(const float* vcand, const float cs
 	// find the threshold hit time to bail out based on the early out penalty
 	// (see how the penalty is calculated below to understnad)
 	float minPen = minPenalty - vpen - vcpen;
-	float tThresold = ((double)m_params.weightToi/(double)minPen - 0.1) * (double)m_params.horizTime;
+	float tThresold = (m_params.weightToi / minPen - 0.1f) * m_params.horizTime;
```

Comment:
Oh my. This comment just made me re-read this (brilliant) article:

https://randomascii.wordpress.com/2012/03/21/intermediate-floating-point-precision/

So... maybe the casts were a good idea, maybe they weren't. Maybe it depends on how the code is compiled, and a hundred other things. I think it's probably best to keep the behaviour the same if we don't know what the impact will be - and just cast the original back to float explicitly.